infra/main.parameters.json

```diff
@@ -45,10 +45,10 @@
       "value": "${AZURE_STORAGE_RESOURCE_GROUP}"
     },
     "chatGptDeploymentName": {
-      "value": "${AZURE_OPENAI_CHATGPT_DEPLOYMENT}"
+      "value": "${AZURE_OPENAI_CHATGPT_DEPLOYMENT=chat}"
```

Comment:
@vhvb1989 This is the functionality I'm wondering about, this way of specifying default values if the env var isn't found.

Reply:
Aren't you setting the default already with `param chatGptDeploymentName string = 'chat'` within the bicep file?

Having the default on bicep means that no matter who deploy the file, the default value will be used.
Having the default value on `main.parameters.json` means that only azd would know about that default value.

I'd go one way or the other, as having both might lead to sync issues, where someone might forget to update both values when updating the default value.

Any other thoughts about this @weikanglim ?

Reply:
The goal is for this to support a few scenarios:

* Devs can specify the parameter by using `azd env set` before running `azd up`, and that gets passed in via main.parameters.json . Just setting `param = 'chat'` doesn't seem to support that scenario - I think because the empty env variable overrides it to an empty string? That's why we have the current workaround with checking if empty() and setting a new variable to 'chat' instead. Is it supposed to work?
* If devs don't specify the parameter, then it is instead set to "chat". That is satisfied by `param = 'chat'` as long as there isn't also main.parameters.json.

Reply:
The syntax works as you expect it @pamelafox. Also agree with @vhvb1989 that it may be best to not have multiple places specifying defaults.

I think it reads better for me if the defaulting + the env var was both defined in `parameters.json` as the single line for the source of truth. This rule also avoids encountering possible errors when the variable isn't a string type.

Reply:
Okay, I've updated this PR to keep the default values in main.parameters.json and I put a comment about that in main.bicep. Makes sense to DRY it up.